modules/twinkleblock.js

```diff
@@ -0,0 +1,1064 @@
+			{ label: 'Generic block', value: 'uw-block' },
+			{ label: 'Generic block – IP', value: 'uw-ablock' },
+			{ label: 'Generic block – indefinite', value: 'uw-blockindef' },
+			{ label: 'Not here to contribute to the encyclopedia', value: 'uw-nothereblock' },
+			{ label: 'Vandalism', value: 'uw-vblock' },
+			{ label: 'Vandalism-only account', value: 'uw-voablock' }
+		],
+	},
+	{
+		label: 'Extended reasons',
+		list: [
+			{ label: 'Advertising', value: 'uw-adblock' },
+			{ label: 'Arbitration enforcement', value: 'uw-aeblock' },
+			{ label: 'Block evasion – IP', value: 'uw-ipevadeblock' },
+			{ label: 'BLP violations', value: 'uw-bioblock' },
+			{ label: 'Copyright violations', value: 'uw-copyrightblock' },
+			{ label: 'Creating inappropriate pages', value: 'uw-npblock' },
+			{ label: 'Disruptive editing', value: 'uw-disruptblock' },
+			{ label: 'Edit filter-related', value: 'uw-efblock' },
+			{ label: 'Edit warring', value: 'uw-ewblock' },
+			{ label: 'Generic block with talk page access revoked', value: 'uw-blocknotalk' },
+			{ label: 'Harassment', value: 'uw-hblock' },
+			{ label: 'Inappropriate use of user talk page while blocked', value: 'blocked talk-revoked-notice' },
+			{ label: 'Legal threats', value: 'uw-lblock' },
+			{ label: 'Personal attacks or harassment', value: 'uw-aoablock' },
+			{ label: 'Possible comprimised account', value: 'uw-compblock' },
+			{ label: 'Removal of content', value: 'uw-dblock' },
+			{ label: 'Sockpuppetry', value: 'uw-spoablock' },
+			{ label: 'Social networking', value: 'uw-myblock' },
+			{ label: 'Spam', value: 'uw-sblock' },
+			{ label: 'Spam/advertising-only account', value: 'uw-soablock' },
+			{ label: 'Unapproved bot', value: 'uw-botblock' },
+			{ label: 'Violating the three-revert rule', value: 'uw-3block' }
+		]
+	},
+	{
+		label: 'Username violations',
+		list: [
+			{ label: 'Bot username', value: 'uw-botublock' },
+			{ label: 'Memorial username soft block', value: 'uw-memorialblock' },
+			{ label: 'Promotional username, hard block', value: 'uw-spamublock' },
+			{ label: 'Promotional username, soft block', value: 'uw-softerblock' },
+			{ label: 'Similar username soft block', value: 'uw-ublock-double' },
+			{ label: 'Username violation, soft block', value: 'uw-ublock' },
+			{ label: 'Username violation, hard block', value: 'uw-uhblock' },
+			{ label: 'Username impersonation hard block', value: 'uw-uhblock-double' },
+			{ label: 'Username represents a famous person, soft block', value: 'uw-ublock-famous' },
+			{ label: 'Username represents a non-profit, soft block', value: 'uw-causeblock' },
+			{ label: 'Username violation, vandalism-only account', value: 'uw-vaublock' }
+		]
+	},
+	{
+		label: 'Templated reasons',
+		list: [
+			{ label: 'blocked proxy', value: 'blocked proxy' },
+			{ label: 'CheckUser block', value: 'CheckUser block' },
+			{ label: 'checkuserblock-account', value: 'checkuserblock-account' },
```

Comment:
Fair enough. Still, I like to see Twinkle improve wherever it can on what 
the default MediaWiki interface offers, not simply copy it, so it would be 
nice to give a more friendly description here.